commands/http/handler.go

```diff
@@ -46,33 +47,61 @@ const (
 	plainText              = "text/plain"
 )
 
+var localhostOrigins = []string{
+	"http://127.0.0.1",
+	"https://127.0.0.1",
+	"http://localhost",
+	"https://localhost",
+}
+
 var mimeTypes = map[string]string{
 	cmds.JSON: "application/json",
 	cmds.XML:  "application/xml",
 	cmds.Text: "text/plain",
 }
 
-func NewHandler(ctx cmds.Context, root *cmds.Command, allowedOrigin string) *Handler {
-	// allow whitelisted origins (so we can make API requests from the browser)
-	if len(allowedOrigin) > 0 {
-		log.Info("Allowing API requests from origin: " + allowedOrigin)
+type ServerConfig struct {
+	// Headers is an optional map of headers that is written out.
+	Headers map[string][]string
+
+	// CORSOpts is a set of options for CORS headers.
+	CORSOpts *cors.Options
+}
+
+func skipAPIHeader(h string) bool {
+	switch h {
+	default:
```

Comment:
seeing default first and implicit continues always weirds me out in the logic. maybe:
```
switch h{
    case "Access-Control-Allow-Origin":
       return true
    case "Access-Control-Allow-Methods":
       return true
    case "Access-Control-Allow-Credentials":
       return true
    default:
        return false
}
```

just to be explicit?